src/core/thread/mesh_forwarder_ftd.cpp

```diff
@@ -244,7 +247,16 @@ otError MeshForwarder::EvictMessage(uint8_t aPriority)
 
     if (error == OT_ERROR_NONE)
     {
-        RemoveMessage(*evict);
+        if (evictQueue == &mSendQueue)
```

Comment:
Good catch or `RemoveMessage()`.

How about asking the message itself about the queue it is in instead of saving the `evictQueue`?
`Message` tracks its own queue and whether it's a normal or a priority queue?

Reply:
Updated as suggested. I had to make `Message::GetPriorityQueue()` public. Also, the current method only supports priority queues for now given that's all it needs to operate on.